crates/store/src/db/mod.rs

```diff
@@ -254,6 +262,53 @@ impl Db {
         db_tx.commit().context("failed to commit database transaction")
     }
 
+    /// Avoid repeated boilerplate, frame the query in a transaction
+    pub(crate) async fn framed<R, E, Q, M>(&self, msg: M, query: Q) -> std::result::Result<R, E>
+    where
+        Q: Send
+            + for<'a, 't> FnOnce(&'a mut Transaction<'t>) -> std::result::Result<R, E>
+            + 'static,
+        R: Send + 'static,
+        M: Send + ToString,
+        E: From<DatabaseError>,
+        E: std::error::Error + Send + Sync + 'static,
+    {
+        let conn = self.pool.get().await.unwrap(); // FIXME XXX TOODO
+
+        conn.interact(|conn| {
+            let mut db_tx = conn.transaction().map_err(DatabaseError::SqliteError)?;
+            let r = query(&mut db_tx)?;
+            db_tx.commit().map_err(DatabaseError::SqliteError)?;
+            Ok(r)
+        })
+        .await
+        .map_err(|err| E::from(DatabaseError::interact(&msg.to_string(), &err)))?
+    }
+
+    /// Avoid repeated boilerplate, frame the query in a transaction
+    pub(crate) async fn framed_without_transaction<R, E, Q, M>(
+        &self,
+        msg: M,
+        query: Q,
+    ) -> std::result::Result<R, E>
+    where
+        Q: Send + FnOnce(&mut Connection) -> std::result::Result<R, E> + 'static,
+        R: Send + 'static,
+        M: Send + ToString,
+        E: From<DatabaseError>,
+        E: From<rusqlite::Error>,
+        E: std::error::Error + Send + Sync + 'static,
+    {
+        let conn = self.pool.get().await.unwrap(); // FIXME XXX TOODO
```

Comment:
Dangling todo

Reply:
This one is still here 
```suggestion
        let conn = self.pool.get().await.map_err(DatabaseError::MissingDbConnection)?;
```